config/go.d/dnsdist.conf

```
@@ -0,0 +1,165 @@
+#    lower numbers make the charts appear before the ones with higher numbers. Default: 70000.
+#
+#
+# [ JOBS ]
+# JOBS allow you to collect values from multiple sources.
+# Each source will have its own set of charts.
+#
+# IMPORTANT:
+#  - Parameter 'name' is mandatory.
+#  - Jobs with the same name are mutually exclusive. Only one of them will be allowed running at any time.
+#
+# This allows autodetection to try several alternatives and pick the one that works.
+# Any number of jobs is supported.
+#
+# The JOBS section format:
+#
+# jobs:
+#   - name: job1
+#     param1: value1
+#     param2: value2
+#
+#   - name: job2
+#     param1: value1
+#     param2: value2
+#
+#   - name: job2
+#     param1: value1
+#
+#
+# [ List of JOB specific parameters ]:
+#
+#  - url
+#    Server URL.
+#    Syntax:
+#      url: http://127.0.0.1:5053
+#
+#  - username
+#    Username for basic HTTP authentication.
+#    Syntax:
+#      username: tony
+#
+#  - password
+#    Password for basic HTTP authentication.
+#    Syntax:
+#      password: stark
+#
+#  - proxy_url
+#    Proxy URL.
+#    Syntax:
+#      proxy_url: http://localhost:3128
+#
+#  - proxy_username
+#    Username for proxy basic HTTP authentication.
+#    Syntax:
+#      username: bruce
+#
+#  - proxy_password
+#    Password for proxy basic HTTP authentication.
+#    Syntax:
+#      username: wayne
+#
+#  - timeout
+#    HTTP response timeout.
+#    Syntax:
+#      timeout: 1
+#
+#  - method
+#    HTTP request method.
+#    Syntax:
+#      method: GET
+#
+#  - body
+#    HTTP body to append.
+#    Syntax:
+#      body: '{fake: data}'
+#
+#  - headers
+#    HTTP request headers.
+#    Syntax:
+#      headers:
+#        X-API-Key: key
+#
+#  - not_follow_redirects
+#    Whether to not follow redirects from the server.
+#    Syntax:
+#      not_follow_redirects: yes/no
+#
+#  - tls_skip_verify
+#    Whether to skip verifying server's certificate chain and hostname.
+#    Syntax:
+#      tls_skip_verify: yes/no
+#
+#  - tls_ca
+#    Certificate authority that client use when verifying server certificates.
+#    Syntax:
+#      tls_ca: path/to/ca.pem
+#
+#  - tls_cert
+#    Client tls certificate.
+#    Syntax:
+#      tls_cert: path/to/cert.pem
+#
+#  - tls_key
+#    Client tls key.
+#    Syntax:
+#      tls_key: path/to/key.pem
+#
+# [ JOB defaults ]:
+#  url: http://127.0.0.1:5053
+#  timeout: 1
+#  method: GET
+#  not_follow_redirects: no
+#  tls_skip_verify: no
+#
+#
+# [ JOB mandatory parameters ]:
+#  - name
+#  - url
+#
+# ------------------------------------------------MODULE-CONFIGURATION--------------------------------------------------
+# [ GLOBAL ]
+# update_every: 1
+# autodetection_retry: 0
+# priority: 70000
+#
+#
+# [ JOBS ]
+#jobs:
+#  - name: local
+#    url: http://127.0.0.1:8083
+#    headers:
+#      X-API-Key: 'dnsdist-api-key' # static pre-shared authentication key for access to the REST API (api-key).
```

```suggestion
jobs:
  - name: local
    url: http://127.0.0.1:8083
    headers:
      X-API-Key: 'dnsdist-api-key' # static pre-shared authentication key for access to the REST API (api-key).
```

Let's have local job active?

---

btw does it work with basic auth? powerdns (auth server) seems to be working only with api-key 🤔  (powerdns recursors works with both).

It it possible to gather the metrics w/o any auth?

---

> btw does it work with basic auth? powerdns (auth server) seems to be working only with api-key thinking (powerdns recursors works with both).

Yes, it works with both `basic_auth` and `api-key`.

> It it possible to gather the metrics w/o any auth?

When you are creating the [webserver](https://dnsdist.org/reference/config.html#webserver), you need to specify both `api key` and `password`, so I do not think it is possible to get data without authentication.

---

it was possible for powerdns_recursor, that is why i asked

---

I tried to remove the user password and I received the following output:

```
bash-5.0$  curl "http://127.0.0.1:8083/jsonstat?command=stats"
<h1>Unauthorized</h1>
bash-5.0$ 
```